File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/SwapFileType.java
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes;

import com.intellij.openapi.fileTypes.LanguageFileType;
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon;
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

public class SwapFileType extends LanguageFileType {
  public static final SwapFileType INSTANCE = new SwapFileType();

  private SwapFileType() {
    super(UnitFileLanguage.INSTANCE);
  }

  @NotNull
  @Override
  public String getName() {
    return "Swap unit configuration for systemd";
  }

  @NotNull
  @Override
  public String getDescription() {
    return "A unit configuration file whose name ends in \".swap\" encodes information about a swap device or file for memory"
           + " paging controlled and supervised by systemd.";
  }

  @NotNull
  @Override
  public String getDefaultExtension() {
    return "swap";
  }

  @Nullable
  @Override
  public Icon getIcon() {
    return UnitFileIcon.FILE;
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/SwapFileType.kt
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes

import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage
import org.jetbrains.annotations.Nls
import javax.swing.Icon

class SwapFileType private constructor() : AbstractUnitFileType(UnitFileLanguage.INSTANCE) {
  override fun getName(): String {
    return "Swap unit configuration for systemd"
  }

  override fun getDescription(): String {
    return displayName
  }

  override fun getDefaultExtension(): String {
    return "swap"
  }

  override fun getIcon(): Icon? {
    return UnitFileIcon.SWAP
  }

  override fun getDisplayName(): @Nls String {
    return "Swap unit configuration (systemd)"
  }

  companion object {
    val INSTANCE = SwapFileType()
  }
}
